README.md

```diff
@@ -1,488 +1,742 @@
+Query engines and other kinds of compilers require numerous tree-like representations of a program, starting with an
+AST. Query engines for example typically parse a query, producing an AST, and then apply multiple passes over the AST to
+*inrecmentally* transform it to a logical plan, then a physical plan, and possibly other intermediate representations.
+Compiler passes and related data structures of this sort are large, complex, and difficult to maintain. PIG's "permuted
+domains" feature increases the maintainability of such compiler passes and related data structures by allowing new type
+domains to be created by specifying only the *differences* to another type domain. This avoids having to duplicate the
+data type definitions that are common to both type domains, allowing more numerous, smaller, less complex and more
+maintainable compiler passes.
 
 PIG's permuted domain feature has been heavily inspired by the [Nanopass Framework](https://nanopass.org/).
 
-## Code Generation
+### Code Generation
 
-PIG generates the following components in Kotlin (and may generate similar components in languages such as Rust in the future): 
+PIG generates the following components in Kotlin (and may generate similar components in languages such as Rust in the
+future):
 
 - Immutable, strongly typed classes representing each data type within each type domain.
 - Abstract base classes for implementing compiler passes that:
-    - Transform from one type domain to another (the developer must only account for the *differences* between the domains!).
+    - Transform from one type domain to another (the developer must only account for the *differences* between the
+      domains!).
     - Transform to a modified tree of the same domain.
 - Functions to convert each tree between instances of the generated classes and its s-expression representation.
 - Builder functions, for easily composing deeply nested instances of the strongly typed classes.
 
 ## API Status
 
-PIG is usable now but its API and the API of the generated Kotlin code is under development and might change.
+PIG is fully feature now but its API and the API of the generated Kotlin code is under development and might change.
```

Comment:
"PIG is fully feature now" wording is slightly confusing. Perhaps "PIG IS full-featured now..." is what you meant?

Reply:
Changed to:
> PIG is mature but its API and the API of the generated Kotlin code is under development and might change.